datafusion/core/src/datasource/memory.rs

```diff
@@ -648,9 +649,14 @@ mod tests {
         // Create a table scan logical plan to read from the source table
         let scan_plan = LogicalPlanBuilder::scan("source", source, None)?.build()?;
         // Create an insert plan to insert the source data into the initial table
-        let insert_into_table =
-            LogicalPlanBuilder::insert_into(scan_plan, "t", &schema, InsertOp::Append)?
-                .build()?;
+        let insert_into_table = LogicalPlanBuilder::insert_into(
+            scan_plan,
+            "t",
+            table_sink,
+            &schema,
```

Comment:
Given the table_source already has the target schema, we could avoid storing it again

Reply:
I was thinking to remove it, but I did not as I got confused about its usage.
looking at the https://github.com/apache/datafusion/blob/3e6d70ee71aa2dc9579b5bcda5e5e5e1cae9a8d2/datafusion/core/src/datasource/listing/table.rs#L2076 I got confused is this scan schema or table schema

Reply:
if it is the scan schema, then it should be possible to get from the `scan_plan` 🤔